src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py

```diff
@@ -331,10 +332,7 @@ def create(
                     temperature=temperature,
                     wavelength=wavelength,
                     position=str(position),
-                    value=try_float(
-                        value,
-                        f"value from block plate {plate_name} and wavelength {wavelength}",
-                    ),
+                    value=try_float_or_none(value) or InvalidJsonFloat.NaN,
```

Comment:
This is fine, but anticipating that we might want to do more of this, how about adding a utility method a la the following and calling it here?
```
def try_float_or_nan(value: Optional[str]) -> JsonFloat:
    if not value:
        return InvalidJsonFloat.NaN
    try:
        return float(value)
    except ValueError:
        return InvalidJsonFloat.NaN
```

Reply:
There are some other spots that use this already, plan to follow up with a util.